opennms-base-assembly/src/main/filtered/bin/opennms

```diff
@@ -305,82 +321,82 @@ doStart(){
 	esac
 
 
-	$JAVA_CMD -Dopennms.home=$OPENNMS_HOME -jar $BOOTSTRAP check
-	if [ $? -ne 0 ]; then
+	if ! "${JAVA_CMD[@]}" "-Dopennms.home=$OPENNMS_HOME" -jar "$BOOTSTRAP" check; then
 		echo "${install.package.description} was unable to connect to the 'opennms' database configured in opennms-datasources.xml." >&2
 		return 1
 	fi
 
 	##########################################################################
 	# Run opennms with the "-t" option to enable the Java Platform Debugging
 	# Architecture. This will open a server socket on port 8001 that can be
-	# connected to by a remote java debugger. A good choice is JSwat which can 
+	# connected to by a remote java debugger. A good choice is JSwat which can
 	# be found at http://www.bluemarsh.com
 	###########################################################################
-	if [ $TEST -gt 0 ]; then
+	if [ "$TEST" -gt 0 ]; then
 		echo "- enabling JPDA debugging on port 8001" >&2
-		JPDA="-agentlib:jdwp=transport=dt_socket,server=y,address=8001,suspend=n"
+		JPDA=("-agentlib:jdwp=transport=dt_socket,server=y,address=8001,suspend=n")
 	fi
 
 	# See: http://www.khelekore.org/jmp/tijmp/
-	if [ $TPROFILE -gt 0 ]; then
+	if [ "$TPROFILE" -gt 0 ]; then
 		echo "- enabling TIJMP Profiling" >&2
-		JPDA="-Dtijmp.jar=/usr/share/java/tijmp-0.6.jar -agentlib:tijmp ${JPDA}"
+		JPDA=("-Dtijmp.jar=/usr/share/java/tijmp-0.6.jar" "-agentlib:tijmp" "${JPDA[@]}")
 	fi
 
 	# See: http://oprofile.sourceforge.net/doc/setup-jit.html
-	if [ $OPROFILE -gt 0 ]; then
+	if [ "$OPROFILE" -gt 0 ]; then
 		echo "- enabling OProfile support" >&2
-		JPDA="-agentpath:/usr/lib/oprofile/libjvmti_oprofile.so ${JPDA}"
+		JPDA=("-agentpath:/usr/lib/oprofile/libjvmti_oprofile.so" "${JPDA[@]}")
 	fi
 
 	if [ "$SERVICE" = "" ]; then
-		APP_VM_PARMS="$JPDA $MANAGER_OPTIONS"
-		APP_PARMS_BEFORE="start"
+		APP_VM_PARMS+=("$MANAGER_OPTIONS")
```

Comment:
This one was copying from the JPDA array before so it can't be simplified to just `+=`
```suggestion
		APP_VM_PARMS=("${JPDA[@]}" "$MANAGER_OPTIONS")
```